arm/Microsoft.ServiceFabric/clusters/deploy.bicep

```diff
@@ -0,0 +1,350 @@
+param reliabilityLevel string = 'None'
+
+@description('Optional. Describes the certificate details.')
+param reverseProxyCertificate object = {}
+
+@description('Optional. Describes a list of server certificates referenced by common name that are used to secure the cluster.')
+param reverseProxyCertificateCommonNames object = {}
+
+@allowed([
+  'Hierarchical'
+  'Parallel'
+])
+@description('Optional. This property controls the logical grouping of VMs in upgrade domains (UDs). This property cannot be modified if a node type with multiple Availability Zones is already present in the cluster.')
+param sfZonalUpgradeMode string = 'Hierarchical'
+
+@description('Optional. Describes the policy used when upgrading the cluster.')
+param upgradeDescription object = {}
+
+@allowed([
+  'Automatic'
+  'Manual'
+])
+@description('Optional. The upgrade mode of the cluster when new Service Fabric runtime version is available.')
+param upgradeMode string = 'Automatic'
+
+@description('Optional. Indicates the end date and time to pause automatic runtime version upgrades on the cluster for an specific period of time on the cluster (UTC).')
+param upgradePauseEndTimestampUtc string = ''
+
+@description('Optional. Indicates the start date and time to pause automatic runtime version upgrades on the cluster for an specific period of time on the cluster (UTC).')
+param upgradePauseStartTimestampUtc string = ''
+
+@allowed([
+  'Wave0'
+  'Wave1'
+  'Wave2'
+])
+@description('Optional. Indicates when new cluster runtime version upgrades will be applied after they are released. By default is Wave0.')
+param upgradeWave string = 'Wave0'
+
+@description('Optional. The VM image VMSS has been configured with. Generic names such as Windows or Linux can be used')
+param vmImage string = ''
+
+@allowed([
+  'Hierarchical'
+  'Parallel'
+])
+@description('Optional. This property defines the upgrade mode for the virtual machine scale set, it is mandatory if a node type with multiple Availability Zones is added.')
+param vmssZonalUpgradeMode string = 'Hierarchical'
+
+@description('Optional. Boolean to pause automatic runtime version upgrades to the cluster.')
+param waveUpgradePaused bool = false
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or it\'s fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'')
+param roleAssignments array = []
+
+@description('Optional. Array of Service Fabric cluster applications.')
+param serviceFabricClusterApplications array = []
```

Comment:
let's keep the name simple like `applications`

Reply:
Changed.